src/RealtimeEffectPanel.cpp

```diff
@@ -420,51 +396,23 @@ namespace
             XO("Remove %s").Format(effectName)
          );
       }
-      
+
       void OnOptionsClicked(wxCommandEvent& event)
       {
          if(mProject == nullptr || mEffectState == nullptr)
             return;//not initialized
-         
+
          const auto ID = mEffectState->GetID();
          const auto effectPlugin = EffectManager::Get().GetEffect(ID);
+
          if(effectPlugin == nullptr)
          {
             ///TODO: effect is not avaialble
             return;
          }
-         auto access = mEffectState->GetAccess();
-         // Copy settings
-         auto initialSettings = access->Get();
-         auto cleanup = EffectManager::Get().SetBatchProcessing(ID);
-
-         std::shared_ptr<EffectInstance> pInstance;
-
-         // Like the call in EffectManager::PromptUser, but access causes
-         // the necessary inter-thread communication of settings changes
-         // if play is in progress
-         bool changed = effectPlugin->ShowHostInterface(
-            ProjectWindow::Get( *mProject), DialogFactory(mEffectState),
-            pInstance, *access, false );
 
-         if (changed)
-         {
-            auto effectName = GetEffectName();
-            ProjectHistory::Get(*mProject).PushState(
-               /*! i18n-hint: undo history record
-                first parameter - realtime effect name
-                second parameter - track name
-                */
-               XO("Changed %s in %s").Format(effectName, mTrack->GetName()),
-               /*! i18n-hint: undo history record
-                first parameter - realtime effect name */
-               XO("Change %s").Format(effectName)
-         );
-         }
-         else
-            // Dialog was cancelled.
-            // Reverse any temporary changes made to the state
-            access->Set(std::move(initialSettings));
```

Comment:
The deleted code for a modal dialog restored state as it was in case the dialog is cancelled or the Esc key is hit, and it pushes an undo history item for change of state if the dialog is OK'd.

I don't see where this commit preserves that behavior by other means.

Reply:
As it was mentioned in PR description - the Undo/Redo is intentionally left unimplemented for the effect parameters. And for the reference - there are several DAWs that do not implement it all as well and at least for Ableton there were problems observed with the implementation (i. e. Undo  restored the parameter value to some unexpected state).

If and when we decide to implement the undo/redo for stacked effect parameters - we will have to create an item for each parameter user changes (see Ableton as the reference). 

The potential problem I expect now is that Audacity project won't get "dirty" on parameter change, but I would rather fix it in a different PR.

Reply:
Was it agreed then with designers that the formerly specified undo/redo behavior is now abandoned?

Do we open another issue for the future effort to figure out the problem?

If there will not be undo/redo states for the OK button, still I think something could be done:  the OK button should call `ModifyState` -- what also happens for the power button changes.  Then the change of settings becomes, not another state, but fused with whatever is the latest undo state.  This is similar to what is done when you simply change the selection, and makes for argubly less surprising behavior, if after OK-ing the dialog, you undo and redo again, expecting the settings to be unchanged.

Besides all that, there is also the reasonable expectation that if you cancel the dialog, tentative changes of settings should be cancelled and old state restored.  But nothing does that now.

Reply:
>Was it agreed then with designers that the formerly specified undo/redo behavior is now abandoned

I've talked with @Tantacrul on that. Can I  see the previous specification please? AFAIK the dialog specified has no Ok/Cancel buttons whatsoever. 

>the OK button

There will be no OK button

Reply:
Please see #3090 for the design. Stackable effects are really different from what happens on destructive path so there are different assumptions about the behavior.

Unfortunately we have a chicken-egg problem here: we cannot reasonably implement #3090  before non modality is achieved and without it there seems to be a confusion about the behavior of the buttons that will be removed from the dialog.

Reply:
So then I looked at slide 8.0 (macOS examples) and that is so.

But is that an omission in the design?  How is the user supposed to close the dialog?  If it's only with Esc or the close button, then there is no way to distinguish between committing and cancelling the changes.

Reply:
I have commented at #3090

Reply:
This matches the behavior for stacked effects in other DAWs though. In most of the cases I saw (including Garage Band) settings are always committed (on every change) and software provides no way to undo them. Ableton tracks each and every parameter change, but every change is committed immediately as well. 

![image](https://user-images.githubusercontent.com/2660628/180988441-71a3d622-6109-4a86-a70e-dbdb3f35122c.png)

Reply:
So maybe we have an opportunity to do things a bit better than those.